### PR TITLE
fix: only auto-attach to a target once

### PR DESCRIPTION
**Pull request by OrKoN**
This can be a workaround for https://github.com/GoogleChromeLabs/chromium-bidi/issues/2420

### CODE REVIEW

File: src/bidiMapper/modules/cdp/CdpTargetManager.ts [hunk 1]
```diff
@@ -154,9 +154,13 @@ export class CdpTargetManager {
...
       case 'page':
       case 'iframe': {
-        if (targetInfo.targetId === this.#selfTargetId) {
+        // Mapper only needs one session per target. If we receive additional
+        // auto-attached sessions, that is very likely coming from custom CDP
+        // sessions.
+        if (this.#targetIdsToBeIgnoredByAutoAttach.has(targetInfo.targetId)) {
           break;
         }
+        // this.#targetIdsToBeIgnoredByAutoAttach.add(targetInfo.targetId);
```
Comment (sadym-chromium) [inline src/bidiMapper/modules/cdp/CdpTargetManager.ts:163]: artefact?

**Review by sadym-chromium — APPROVED** [review]

**Comment by OrKoN** [comment]
@sadym-chromium not ready for review

**Comment by sadym-chromium** [comment]
> @sadym-chromium not ready for review

Can you please mark it as draft?

File: src/bidiMapper/modules/cdp/CdpTargetManager.ts [hunk 2]
```diff
@@ -154,9 +156,18 @@ export class CdpTargetManager {
...
       case 'iframe': {
-        if (targetInfo.targetId === this.#selfTargetId) {
+        if (this.#selfTargetId === targetInfo.targetId) {
+          // break to detach from the self target.
           break;
         }
+        // Mapper only needs one session per target. If we receive additional
+        // auto-attached sessions, that is very likely coming from custom CDP
+        // sessions.
+        if (this.#targetIdsToBeIgnoredByAutoAttach.has(targetInfo.targetId)) {
```
Comment (Lightning00Blade) [inline src/bidiMapper/modules/cdp/CdpTargetManager.ts:166]: Why are we doing this only for these type of targets?

Comment (OrKoN) [reply src/bidiMapper/modules/cdp/CdpTargetManager.ts:166]: good point, we probably can do it for all sessions

File: .github/workflows/puppeteer.yml [hunk 3]
```diff
@@ -30,8 +30,6 @@ jobs:
         uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
         with:
           path: 'bidi'
-      - name: Set up FFmpeg
```
Comment (OrKoN) [inline .github/workflows/puppeteer.yml:33]: not actually needed and often fails

File: tests/session/test_cdp.py [hunk 4]
```diff
@@ -158,3 +159,52 @@ async def test_cdp_wait_for_event(websocket, get_cdp_session_id, context_id):
...
+
+    session_id = None
+    with pytest.raises(asyncio.TimeoutError):
+        while True:
+            if 'id' in event and event['id'] == id:
+                session_id = event['result']['result']['sessionId']
+            if 'id' not in event:
+                events.append(event)
+            event = await asyncio.wait_for(read_JSON_message(websocket),
+                                           timeout=1.0)
```
Comment (sadym-chromium) [inline tests/session/test_cdp.py:205]: Can you send another command an verify it is finished with no extra messages instead of waiting for 1 second?

Comment (OrKoN) [reply tests/session/test_cdp.py:205]: Could you post a suggestion in code? I didn't get the idea

Comment (sadym-chromium) [reply tests/session/test_cdp.py:206]: https://github.com/GoogleChromeLabs/chromium-bidi/commit/e864c4cfb59057ad29533e787df19c9885c810d1

Comment (OrKoN) [reply tests/session/test_cdp.py:206]: The timeout was not the cause for the issue, I have updated the PR to reflect the fix.

File: tools/bidi-server.mjs [hunk 5]
```diff
@@ -55,7 +55,7 @@ export function createLogFile(suffix) {
 export function parseCommandLineArgs() {
   return yargs(hideBin(process.argv))
     .usage(
-      `$0 [fileOrFolder]`,
+      `$0 [fileOrFolder...]`,
```
Comment (OrKoN) [inline tools/bidi-server.mjs:58]: drive by to support multiple test files

File: src/bidiMapper/modules/cdp/CdpTargetManager.ts [hunk 6]
```diff
@@ -149,20 +151,42 @@ export class CdpTargetManager {
     parentSessionCdpClient: CdpClient
   ) {
     const {sessionId, targetInfo} = params;
+
+    if (
+      this.#selfTargetId !== targetInfo.targetId &&
+      targetInfo.type !== 'service_worker'
+    ) {
+      // Service workers are special case because they attach to the
```
Comment (Lightning00Blade) [inline src/bidiMapper/modules/cdp/CdpTargetManager.ts:159]: Left over comment?

**Review by Lightning00Blade — APPROVED** [review]
LGTM

File: src/bidiMapper/modules/cdp/CdpTargetManager.ts (hunk 6 above)
Comment (OrKoN) [reply src/bidiMapper/modules/cdp/CdpTargetManager.ts:167]: no, it still applies to the service worker check

Comment (OrKoN) [reply src/bidiMapper/modules/cdp/CdpTargetManager.ts:167]: I have updated it again to fix WPT

File: src/bidiMapper/modules/cdp/CdpTargetManager.ts [hunk 7]
```diff
@@ -151,18 +153,50 @@ export class CdpTargetManager {
...
+    }
+
     switch (targetInfo.type) {
       case 'page':
       case 'iframe': {
-        if (targetInfo.targetId === this.#selfTargetId) {
-          break;
+        if (this.#selfTargetId === targetInfo.targetId) {
+          void detach();
+          return;
```
Comment (OrKoN) [inline src/bidiMapper/modules/cdp/CdpTargetManager.ts:192]: I have removed breaks leading to detach at the end with an explicit detach

**Pull request merged**